hrplib/hrpModel/ForwardDynamics.cpp

```diff
@@ -66,6 +66,27 @@ void ForwardDynamics::enableSensors(bool on)
 }
 
 
+double ForwardDynamics::jointFriction(Link* link,
+                                      double clmb_threshold,
+                                      unsigned int poly) const
+{
+  double signDq;
+  if(link->dq>clmb_threshold) signDq=1.0;
+  else if(link->dq<-clmb_threshold) signDq=-1.0;
+  else{
+    signDq = link->dq/clmb_threshold;
+    if (poly == 1);
+    else if (poly == 2 && link->dq >  0) signDq = signDq*signDq;
+    else if (poly == 2 && link->dq <= 0) signDq = -signDq*signDq;
+    else signDq = signDq*signDq*signDq;
+  }
+  //Smoothly set Friction Coefficients according to velocity sign
+  double Kv = 0.5* ( link->Kv_p * (1+signDq) + link->Kv_n * (1-signDq) );
+  double Ka = 0.5* ( link->Ka_p * (1+signDq) + link->Ka_n * (1-signDq) );
+  double Kf = 0.5* ( link->Kf_p * (1+signDq) + link->Kf_n * (1-signDq) );
+  return Kv*link->dq + Ka*link->ddq + Kf*signDq;
```

Comment:
Dear Rohan,
Thanks for this interesting commit. It is going in the good direction
Could you elaborate on the model you are using ?
If possible put the name of the model you are using and the reference you are using.
Several physical phenomena can be taken into account and it is better for every one to be clear on this part.

Otherwise as stated in the summary touching at the simulator itself requires (IMHO) a unitary test to keep confidence on the output of this piece of software. Or please make it clear in the code that the friction can be disabled. (not set to zero but clearly disabled)
Thanks again for your time and your proposal.

Reply:
Hi Rohan, a couple of things:
1) You shouldn't use the term Ka*ddq because that's not a friction: that's supposed to compensate for the acceleration of the robot inertia, but actually we are not using it on the robot (i.e. Ka=0) 
2) Use a standard sign function for computing signDq: the smooth sign is necessary for compensation to avoid discontinuities in the control, but the real friction is indeed discontinuous at zero velocity

Reply:
Hi Rohan and Andrea,
After mulling over it, it could be preferable to have a separate copy of the hrplib running on a different dynamic simulator CORBA server. This way you could develop and checked your own code while others (like me) can continue to use the server that has been thoroughly tested. This will have the additional value to be able to make comparisons without handling installations of different version. What do you think ?